### ISSUE TITLE
Getting a "UnboundLocalError" when trying to load data. 

### ISSUE THREAD

**Issue by Zaher503** [dattalab/keypoint-moseq#157]
Hey, 

I am using KPMS on some .slp files. I followed the [Project setup](https://keypoint-moseq.readthedocs.io/en/latest/modeling.html) guide up to the Load data section where I would get the following error:

```
# load data
keypoint_data_path = data_dir # can be a file, a directory, or a list of files
coordinates, confidences, bodyparts = kpms.load_keypoints(keypoint_data_path, 'sleap')

# format data for modeling
data, metadata = kpms.format_data(coordinates, confidences, **config())
```

```
Loading keypoints:   0%|                         | 0/48 [00:00<?, ?it/s]

Error loading ../metallo-
pilot.220809_110605/Camera0/mov.000014.predictions.proofread.slp:
[Errno 22] Invalid argument: '../metallo-
pilot.220809_110605/Camera0/Camera0\\mov.00014.mp4'


---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
Cell In[5], line 3
      1 # load data
      2 keypoint_data_path = data_dir # can be a file, a directory, or a list of files
----> 3 coordinates, confidences, bodyparts = kpms.load_keypoints(keypoint_data_path, 'sleap')
      5 # format data for modeling
      6 data, metadata = kpms.format_data(coordinates, confidences, **config())

File /opt/conda/envs/keypoint_moseq/lib/python3.9/site-packages/keypoint_moseq/io.py:1044, in load_keypoints(filepath_pattern, format, extension, recursive, path_sep, path_in_name, remove_extension, exclude_individuals)
   1041     except Exception as e:
   1042         print(fill(f"Error loading {filepath}: {e}"))
-> 1044     coordinates.update(new_coordinates)
   1045     confidences.update(new_confidences)
   1047 # check for valid results

UnboundLocalError: local variable 'new_coordinates' referenced before assignment
```

I am using version 0.4.7 on JupyterLab. Please let me know if I can provide more information. 

Screenshot of the error I am getting:
![image](https://github.com/user-attachments/assets/d0a275fe-5478-43e9-ba38-7d40276eacaf)

**Comment by talmo** [comment]
Cross linking: this is a bug on our end (https://github.com/talmolab/sleap-io/issues/106#issuecomment-2257081774) related to path sanitization on weird platforms that don't like pathlib.

Issue on this end is that the `try-except` block in the data loader doesn't re-raise the exception:

https://github.com/dattalab/keypoint-moseq/blob/1e2596559ff1ddaa42f81cd15de3287e0271d5a8/keypoint_moseq/io.py#L1024-L1044

If an exception occurred in L1025-1026, when you get to L1044, `new_coordinates` won't have been defined.

Maybe restrict the except block to just the `ValueError` that you're raising, or otherwise re-raise the exception after logging?

In any case, we'll push a fix to this on the `sleap-io` side soon. Shouldn't change anything on the KPMS side.

**Comment by talmo** [comment]
Fixed in [sleap-io v0.1.7](https://github.com/talmolab/sleap-io/releases/tag/v0.1.7).

**Comment by calebweinreb** [comment]
Thanks for being on top of this Talmo! I totally agree that the error logic here needs improvement. I'm going to keep this issue open while we work on that.
